### PR TITLE
Routing and extraction in Tide

### PR DESCRIPTION
**Pull request by aturon**
Next Tide blog post!

[Rendered](https://github.com/rust-lang-nursery/wg-net/blob/605e7412236baa31cc13546a70c628ce25f42dfd/_posts/2018-10-16-tide-routing.md)

### CODE REVIEW

**Comment by aturon** [comment]
Note: the link to the internals post is intentionally "broken"; I'll fill it in after creating a thread.

File: _posts/2018-10-16-tide-routing.md [hunk 1]
```diff
@@ -0,0 +1,382 @@
+---
+layout: post
+title: "Routing and extraction in Tide: a first sketch"
+date: 2018-10-16
+author: Aaron Turon
+---
+
+This post continues the [series on Tide][last-post], sketching a *possible* design for routing and extraction that combines some of the best ideas from frameworks like Rocket, Actix, and Gotham.
```
Comment (killercup) [inline _posts/2018-10-16-tide-routing.md]:
```suggestion
This post continues the [series on Tide][last-post], sketching a *possible* design for routing and extraction that combines some of the best ideas from frameworks like [Rocket], [Actix], and [Gotham].

[Rocket]: https://rocket.rs/
[Actix]: https://actix.rs/
[Gotham]: https://gotham.rs/
```

File: _posts/2018-10-16-tide-routing.md [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+    /// Attempt to look up a message by ID
+    fn get(&self, id: u64) -> Option<Message>;
+
+    /// Attempt to edit a message; returns `false`
+    /// if `id` is not found.
+    fn set(&mut self, id: u64, msg: Message) -> bool;
+}
+```
+
+This tiny API is meant as a stand-in for more complex back ends. The main point of interest is that `Database` a *handle* to a database, meaning that it is `Clone` (and uses an `Arc` under the hood). We'll see why that's important later on.
```
Comment (killercup) [inline _posts/2018-10-16-tide-routing.md]:
```suggestion
This tiny API is meant as a stand-in for more complex back ends. The main point of interest is that `Database` is a *handle* to a database, meaning that it is `Clone` (and uses an `Arc` under the hood). We'll see why that's important later on.
```

File: _posts/2018-10-16-tide-routing.md [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+
+## The web API: table of contents
```
Comment (killercup) [inline _posts/2018-10-16-tide-routing.md]:
```suggestion
## The web API: Table of contents
```

File: _posts/2018-10-16-tide-routing.md [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+The table of contents for the app is specified via a builder API:
+
+```rust
+fn main() {
+    // The endpoints will receive a handle to the app state, i.e. a `Database` instance
+    let mut app = App::new(Database::new());
+
+    app.at("/message").post(new_message);
+    app.at("/message/{}").get(get_message);
+    app.at("/message/{}").post(set_message);
```
Comment (killercup) [inline _posts/2018-10-16-tide-routing.md]: consider using `patch` or `put` instead of `post`

**Pull request merged**